openedx/adg/lms/applications/managers.py

```diff
@@ -17,26 +18,97 @@ def get_queryset(self):
         )
 
 
-class PrerequisiteCourseGroupManager(Manager):
+class MultilingualCourseGroupManager(Manager):
     """
-    Manager which returns all non-empty pre requisite groups
+    Manager for MultilingualCourseGroup
     """
 
-    def get_queryset(self):
-        return super().get_queryset().filter(
-            is_prerequisite=True,
-            multilingual_courses__isnull=False
-        ).distinct()
+    def prereq_course_groups(self):
+        """
+        Get non-empty prerequisite course groups
+        """
+        return self.get_queryset().filter(is_prerequisite=True, multilingual_courses__isnull=False).distinct()
+
+    def get_courses(self, user, is_prereq=False):
+        """
+        Get courses from course groups.
+        Following are the preferences for the course list.
+
+        1. Enrollment
+            If a user is enrolled in any of the courses of a course
+            group then that course is selected from the group.
+
+        2. Language preferred
+            If user has not enrolled in any of the courses of a course
+            group then find a course with preferred language.
+
+        3. First Course
+            If user enrolled and preferred language courses are not found then
+            return the first course of a group.
+
+        Args:
+            is_prereq (bool):  List of MultilingualCourseGroups
+            user (User): user for which we need to find courses
+
+        Returns:
+            list: List of courses which contains a course from each group
+        """
+        courses_list = []
+        course_groups = self.prereq_course_groups() if is_prereq else self.get_queryset()
+
+        for course_group in course_groups:
+            open_multilingual_courses = course_group.multilingual_courses.open_multilingual_courses()
+            enrolled_course = open_multilingual_courses.get_enrolled_course(user) if not user.is_anonymous else None
+            multilingual_course = (
+                enrolled_course or
+                open_multilingual_courses.get_preferred_lang_course() or
+                open_multilingual_courses.first()
```

Comment:
These operations are more relevant to `MultilingualCourseQuerySet`, How about moving this get multilingual course logic there and docstring also